benchmarker/cmd/benchmark_run.go

```diff
@@ -14,10 +15,123 @@ import (
 	"strings"
 	"sync"
 	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	weaviategrpc "github.com/weaviate/weaviate/grpc"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
 )
 
-func benchmark(cfg Config, getQueryFn func(className string) []byte) Results {
+type QueryWithNeighbors struct {
+	Query     []byte
+	Neighbors []int
+}
+
+func processQueueHttp(queue []QueryWithNeighbors, cfg *Config, c *http.Client, m *sync.Mutex, times *[]time.Duration) {
+	for _, query := range queue {
+		r := bytes.NewReader(query.Query)
+		before := time.Now()
+		var url string
+		if cfg.API == "graphql" {
+			url = cfg.Origin + "/v1/graphql"
+		} else if cfg.API == "rest" {
+			url = fmt.Sprintf("%s/v1/objects/%s/_search", cfg.Origin, cfg.ClassName)
+		}
+		req, err := http.NewRequest("POST", url, r)
+		if err != nil {
+			fmt.Printf("ERROR: %v\n", err)
+			continue
+		}
+
+		req.Header.Set("content-type", "application/json")
+
+		if cfg.HttpAuth != "" {
+			req.Header.Set("Authorization", fmt.Sprintf("Bearer %s", cfg.HttpAuth))
+		}
+
+		res, err := c.Do(req)
+		if err != nil {
+			fmt.Printf("ERROR: %v\n", err)
+			continue
+		}
+		took := time.Since(before)
+		bytes, _ := ioutil.ReadAll(res.Body)
+		res.Body.Close()
+		var result map[string]interface{}
+		if err := json.Unmarshal(bytes, &result); err != nil {
+			fmt.Printf("JSON error: %v\n", err)
```

Comment:
If we ignore the error here, line 66 and 74 will panic because result is not instantiated.